README.md

```diff
@@ -1,7 +1,154 @@
 # Go Obscuro
-This is the reference implementation of the Obscuro Protocol.
 
-## Structure
+This repository contains the reference implementation of the [Obscuro Protocol](https://whitepaper.obscu.ro/).
+
+
+## High level overview
+
+The typical blockchain node runs multiple services in a single process. For example:
+- P2P Service
+- RPC Service
+- Data storage
+- Transaction execution
+- Mempool
+
+Obscuro uses TEEs (Intel SGX) to execute transactions in a confidential environment, which means we diverge from the typical architecture. 
+There are three main components of the architecture, each running as a separate process: the enclave, the host and the wallet extension.
+
+[//]: # (Todo - @joel - change the diagram to represent the wallet extension)
+![Architechture](design/obscuro_arch.jpeg)
+
+### I. The Enclave
+
+This is the core component of Obscuro which runs inside the TEE. 
+See [go/enclave](go/enclave)
+
+We use [EGo](https://www.edgeless.systems/products/ego/), an open source SDK for developing this confidential component.
+
+The Enclave exposes an [interface](go/common/enclave.go) over RPC which attempts to minimise the TCB (Trusted computing base).
+
+The enclave component has these main responsibilities:
+
+#### 1. Execute EVM transactions
+Obscuro has the goal to be fully compatible with the EVM, so smart contracts can be ported freely from other EVM compatible
+chains. To achieve this and minimise the effort and incompatibilities, we depend on [go-ethereum](https://github.com/ethereum/go-ethereum).
+
+The dependency on go-ethereum is not straight forward, since transaction execution is coupled with ethereum specific consensus rules,
+which had to be mocked out.
+
+See [go/enclave/evm](go/enclave/evm)
+
+
+#### 2. Store the state
+The dependency on go-ethereum for transaction execution, means that we have to use the same storage interfaces.
```

Comment:
"we have to use the same" or just "we use the same". Have to sounds we're reluctant to use the same storage interfaces.

Reply:
freudian slip :)